datascience/tables.py

```diff
@@ -1328,98 +1328,110 @@ def with_rows(self, rows):
+                value in the new column is ``values``. If sequence of values, 
+                new column takes on values in ``values``. 
 
                 If a sequence, the new column contains the values in
-                ``values``. ``values`` must be the same length as the table.
+                ``values``, must be the same length as the table.
 
         Raises:
             ``ValueError``: If
                 - ``label`` is not a valid column name
-                - ``values`` is a list/array and does not have the same length
+                - if ``label`` is not of type (str) 
+                - ``values`` is a list/array that does not have the same length
                   as the number of rows in the table.
 
-        >>> tiles = Table().with_columns([
-        ...     'letter', ['c', 'd'],
-        ...     'count',  [2, 4],
-        ... ])
-        >>> tiles.with_column('points', [3, 2])
-        letter | count | points
-        c      | 2     | 3
-        d      | 4     | 2
-        >>> tiles.with_column('count', 1)
+        Returns:
+            copy of original table with new or replaced column
+
+        >>> alphabet
+        letter | count 
+        c      | 2     
+        d      | 4     
+        >>> alphabet.with_column('permutes', make_array('a', 'g')) 
+        letter | count | permutes
+        c      | 1     | a
+        d      | 2     | g
+        >>> alphabet
+        letter | count
+        c      | 2
+        d      | 4
+        >>> alphabet.with_column('count', 1)
         letter | count
         c      | 1
         d      | 1
+        >>> alphabet.with_column(1, make_array(1, 2))
+        Traceback (most recent call last):
+            ...
+        ValueError: The column label must be a string, but a int was given
+        >>> alphabet.append_column('bad_col', make_array(1))
+        Traceback (most recent call last):
+            ...
+        ValueError: Column length mismatch. New column does not have the same number of rows as table.
         """
         new_table = self.copy()
         new_table.append_column(label, values)
         return new_table
 
     def with_columns(self, *labels_and_values):
-        """Return a table with additional or replaced columns.
+        """Return a table with additional or replaced columns. 
+            
 
         Args:
-            ``labels_and_values``: An alternating list of labels and values or
-                a list of label-values pairs.
+            ``labels_and_values``: An alternating list of labels and values or a list of label-values pairs. 
+            Values can be any combination of sequences and single values. If one of the labels is in existing table, 
+            the existing column is replaced by its values pair. If any of the values is just a single value (``int``), 
+            every value in the corresponding column is that value. Columns arranged in order of labels as they appear in 
+            ``labels_and_values``.
```

Comment:
nit: Please keep lines to under 79 characters (https://www.python.org/dev/peps/pep-0008/#maximum-line-length). Also, could you indent this text one level like the method you changed above?

Let's just take out the "Values can be any combination of sequences and single values." since it's confusing.

> If one of the labels is in existing table, the existing column is replaced by its values ~~pair~~.

How about:
> ...every value in the corresponding column is set to that value.

And:
> Columns added in order of labels